### PR TITLE
[Meson] Update the build system for CUTEST

### PR DESCRIPTION
**Pull request by amontoison**
#187

### CODE REVIEW

**Comment by amontoison** [comment]
@nimgould @jfowkes 
I tried to compile GALAHAD with CUTEST tonight and it seems that the files related to CUTEST interfaces generate Fortran modules with identical names.
Meson complains about it.
```
meson setup builddir_cutest -Dlibcutest_single_path=/home/alexis/Bureau/git/CUTEst/objects/pc64.lnx.gfo/single \
                                                    -Dlibcutest_double_path=/home/alexis/Bureau/git/CUTEst/objects/pc64.lnx.gfo/double \
                                                    -Dlibcutest_single_modules=/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single \
                                                    -Dlibcutest_double_modules=/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/double

**Comment by amontoison** [comment]
The bug was between the chair and the screen... I forgot to not compile the dummy cutest file in the `dum` folder when we find CUTEST.
I have the following error now:
```
[732/3009] Compiling Fortran object libgalahad_si.../meson-generated_single_run_lancelot_simple.f90.o
FAILED: libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o -c libgalahad_single.so.p/single_run_lancelot_simple.f90
../src/lancelot_simple/run_lancelot_simple.F90:23:9:

   23 |     USE LANCELOT_PRECISION
      |         1
Fatal Error: Cannot open module file ‘lancelot_precision.mod’ for reading at (1): Aucun fichier ou dossier de ce type
compilation terminated.
[734/3009] Compiling Fortran object libgalahad_si...meson-generated_single_uselancelot_steering.f90.o
FAILED: libgalahad_single.so.p/meson-generated_single_uselancelot_steering.f90.o libgalahad_single.so.p/galahad_uselancelot_steering_single.mod 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_uselancelot_steering.f90.o -c libgalahad_single.so.p/single_uselancelot_steering.f90
../src/lancelot/uselancelot_steering.F90:429:55:

  429 |        CALL SPECFILE_assign_logical( spec( 11 ), testal, errout )
      |                                                       1
Error: Symbol ‘testal’ at (1) has no IMPLICIT type; did you mean ‘ttotal’?
../src/lancelot/uselancelot_steering.F90:844:13:

  844 |              RANGE , inform%status, DRCHE_S, ELFUN  = ELFUN  )
      |             1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:850:13:

  850 |              RANGE , inform%status, DRCHE_S )
      |             1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1025:19:

 1025 |              prob, RANGE , data, inform%ncalcf, ICALCF, FT, GVALS, FUVALS,     &
      |                   1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1031:19:

 1031 |              prob, RANGE , data, inform%ncalcf, ICALCF, FT, GVALS, FUVALS,     &
      |                   1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1037:19:

 1037 |              prob, RANGE , data, inform%ncalcf, ICALCF, FT, GVALS, FUVALS,     &
      |                   1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1043:19:

 1043 |              prob, RANGE , data, inform%ncalcf, ICALCF, FT, GVALS,  FUVALS,    &
      |                   1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1049:19:

 1049 |              prob, RANGE , data, inform%ncalcf, ICALCF, FT, GVALS, FUVALS,     &
      |                   1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1113:17:

 1113 |            prob, RANGE , GVALS , FT, XT, FUVALS, lfuval, ICALCF, ICALCG, IVAR, &
      |                 1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1117:17:

 1117 |            prob, RANGE , GVALS , FT, XT, FUVALS, lfuval, ICALCF, ICALCG, IVAR, &
      |                 1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1121:17:

 1121 |            prob, RANGE , GVALS , FT, XT, FUVALS, lfuval, ICALCF, ICALCG, IVAR, &
      |                 1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
../src/lancelot/uselancelot_steering.F90:1125:17:

 1125 |            prob, RANGE , GVALS , FT, XT, FUVALS, lfuval, ICALCF, ICALCG, IVAR, &
      |                 1
Error: Interface mismatch in dummy procedure ‘range’ at (1): INTENT mismatch in argument 'w2'
```

**Comment by nimgould** [comment]
I fixed the above grumbles (I hope!)

**Comment by jfowkes** [comment]
@amontoison please rebase master to pick up the LANCELOT bug fixes (we can't do this as you're using a branch on your fork).

**Comment by amontoison** [comment]
@nimgould 
```
FAILED: libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o -c libgalahad_single.so.p/single_run_lancelot_simple.f90
../src/lancelot_simple/run_lancelot_simple.F90:28:21:

   28 |     REAL ( KIND = rp_ ), PARAMETER :: infinity = 10.0_rp_ ** 20
      |                     1
Error: Symbol ‘rp_’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:29:24:

   29 |     INTEGER ( KIND = ip_ ) :: n, neq, nin, iters, maxit, print_level, exit_code
      |                        1
Error: Symbol ‘ip_’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:30:21:

   30 |     REAL ( KIND = rp_ ) :: gradtol, feastol, fx
      |                     1
Error: Symbol ‘rp_’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:32:21:

   32 |     REAL ( KIND = rp_ ),     ALLOCATABLE, DIMENSION(:) :: BL, BU, X, CX, Y
      |                     1
Error: Symbol ‘rp_’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:43:14:

   43 |     ALLOCATE( X( n  ), BL( n ), BU( n ), CX( neq+nin ), Y( neq+nin ) )
      |              1
Error: Allocate-object at (1) is neither a data pointer nor an allocatable variable
../src/lancelot_simple/run_lancelot_simple.F90:56:7:

   56 |     BU(2) =  3.0_rp_
      |       1
Error: ‘bu’ at (1) is not a variable
../src/lancelot_simple/run_lancelot_simple.F90:68:30:

   68 |     gradtol     = 0.00001_rp_       ! identical to default
      |                              1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:69:30:

   69 |     feastol     = 0.00001_rp_       ! identical to default
      |                              1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:84:18:

   84 |     DEALLOCATE( X, BL, BU, CX, Y )
      |                  1
Error: Allocate-object at (1) is not a nonprocedure pointer nor an allocatable variable
../src/lancelot_simple/run_lancelot_simple.F90:93:17:

   93 |     REAL( KIND = rp_ ), INTENT( IN )   :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:94:17:

   94 |     REAL( KIND = rp_ ), INTENT( OUT )  :: F
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:95:21:

   95 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL   :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:99:21:

   99 |         F = 100.0_wp*(X(2)-X(1)**2)**2 +(1.0_wp-X(1))**2                      !
      |                     1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:106:32:

  106 |                 F = X(1)+3.0_wp*X(2)-3.0_rp_                                   !
      |                                1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:111:44:

  111 |                 F = X(1)**2+X(2)**2-4.0_rp_                                    !
      |                                            1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:121:17:

  121 |     REAL( KIND = rp_ ), INTENT( IN )  :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:122:17:

  122 |     REAL( KIND = rp_ ), INTENT( OUT ) :: G( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:123:21:

  123 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL  :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:152:17:

  152 |     REAL( KIND = rp_ ), INTENT( IN )  :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:153:17:

  153 |     REAL( KIND = rp_ ), INTENT( OUT ) :: H( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:154:21:

  154 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL  :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:78:33:

   78 |                           CX = CX, Y = Y, ITERS  = iters, MAXIT = maxit,       &
      |                                 1
Error: Symbol ‘cx’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:74:46:

   74 |     CALL LANCELOT_simple( n,  X, fx, exit_code,                                &
      |                                              1
Error: Symbol ‘exit_code’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:79:62:

   79 |                           GRADTOL = gradtol, FEASTOL = feastol,                &
      |                                                              1
Error: Symbol ‘feastol’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:74:35:

   74 |     CALL LANCELOT_simple( n,  X, fx, exit_code,                                &
      |                                   1
Error: Symbol ‘fx’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:79:43:

   79 |                           GRADTOL = gradtol, FEASTOL = feastol,                &
      |                                           1
Error: Symbol ‘gradtol’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:54:21:

   54 |     BL(2) = -infinity
      |                     1
Error: Symbol ‘infinity’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:78:56:

   78 |                           CX = CX, Y = Y, ITERS  = iters, MAXIT = maxit,       &
      |                                                        1
Error: Symbol ‘iters’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:67:9:

   67 |     maxit       = 100
      |         1
Error: Symbol ‘maxit’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:37:5:

   37 |     n   = 2  ! number of variables
      |     1
Error: Symbol ‘n’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:38:7:

   38 |     neq = 1  ! number of equality constraints, excluding fixed variables
      |       1
Error: Symbol ‘neq’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:39:7:

   39 |     nin = 1  ! number of inequality (<= 0) constraints, excluding bounds
      |       1
Error: Symbol ‘nin’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:70:15:

   70 |     print_level = 1                ! identical to default
      |               1
Error: Symbol ‘print_level’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:74:31:

   74 |     CALL LANCELOT_simple( n,  X, fx, exit_code,                                &
      |                               1
Error: Symbol ‘x’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:78:40:

   78 |                           CX = CX, Y = Y, ITERS  = iters, MAXIT = maxit,       &
      |                                        1
Error: Symbol ‘y’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:54:4:

   54 |     BL(2) = -infinity
      |    1
Error: Function ‘bl’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:55:4:

   55 |     BU(1) =  infinity              ! upper bounds (componentwise)
      |    1
Error: Function ‘bu’ at (1) has no IMPLICIT type
../src/lancelot_simple/run_lancelot_simple.F90:96:24:

   96 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
../src/lancelot_simple/run_lancelot_simple.F90:124:24:

  124 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
../src/lancelot_simple/run_lancelot_simple.F90:155:24:

  155 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
```

**Comment by jfowkes** [comment]
@amontoison please rebase master to get the latest LANCELOT fix.

**Comment by amontoison** [comment]
@jfowkes @nimgould 
```
FAILED: libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_run_lancelot_simple.f90.o -c libgalahad_single.so.p/single_run_lancelot_simple.f90
../src/lancelot_simple/run_lancelot_simple.F90:94:17:

   94 |     REAL( KIND = rp_ ), INTENT( IN )   :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:95:17:

   95 |     REAL( KIND = rp_ ), INTENT( OUT )  :: F
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:96:21:

   96 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL   :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:100:21:

  100 |         F = 100.0_wp*(X(2)-X(1)**2)**2 +(1.0_wp-X(1))**2                      !
      |                     1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:107:32:

  107 |                 F = X(1)+3.0_wp*X(2)-3.0_rp_                                   !
      |                                1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:112:44:

  112 |                 F = X(1)**2+X(2)**2-4.0_rp_                                    !
      |                                            1
Error: Missing kind-parameter at (1)
../src/lancelot_simple/run_lancelot_simple.F90:122:17:

  122 |     REAL( KIND = rp_ ), INTENT( IN )  :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:123:17:

  123 |     REAL( KIND = rp_ ), INTENT( OUT ) :: G( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:124:21:

  124 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL  :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:153:17:

  153 |     REAL( KIND = rp_ ), INTENT( IN )  :: X( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:154:17:

  154 |     REAL( KIND = rp_ ), INTENT( OUT ) :: H( : )
      |                 1
Error: Parameter ‘rp_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:155:21:

  155 |     INTEGER ( KIND = ip_ ), INTENT( IN ), OPTIONAL  :: i
      |                     1
Error: Parameter ‘ip_’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
../src/lancelot_simple/run_lancelot_simple.F90:81:53:

   81 |                           PRINT_LEVEL = print_level )
      |                                                     1
Error: Type mismatch in argument ‘my_fun’ at (1); passed REAL(4) to UNKNOWN
../src/lancelot_simple/run_lancelot_simple.F90:97:24:

   97 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
../src/lancelot_simple/run_lancelot_simple.F90:125:24:

  125 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
../src/lancelot_simple/run_lancelot_simple.F90:156:24:

  156 |     IF ( .NOT. PRESENT( i ) ) THEN
      |                        1
Error: ‘a’ argument of ‘present’ intrinsic at (1) must be of an OPTIONAL dummy variable
```

**Comment by jfowkes** [comment]
@nimgould we're still seeing LANCELOT compile failures (see above).

**Comment by amontoison** [comment]
Jari, I pushed my branch `cutest` on `ralna/GALAHAD`.
You can do local and I will try to add CI with CUTEST this morning in the train for Argonne.

**Comment by jfowkes** [comment]
Amazing thank you!

**Comment by nimgould** [comment]
Sorry, my blunder. Now fixed

**Comment by amontoison** [comment]
@nimgould 
```
../src/lancelot_simple/run_lancelot_simple.F90:81:53:

   81 |                           PRINT_LEVEL = print_level )
      |                                                     1
Error: Type mismatch in argument ‘my_fun’ at (1); passed REAL(4) to UNKNOWN
```
I tried to use the flag `-fallow-argument-mismatch` but it's not working.

File: .github/workflows/standalone.yml [hunk 1]
```diff
@@ -47,7 +47,13 @@ jobs:
 
       - name: Set the environment variable GALAHAD
         shell: bash
-        run: echo "GALAHAD=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+        run: |
+          echo "GALAHAD=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "DEPS="$GITHUB_WORKSPACE/.." >> $GITHUB_ENV
```
Comment (amontoison) [inline .github/workflows/standalone.yml:52]:
```suggestion
          echo "DEPS=$GITHUB_WORKSPACE/.." >> $GITHUB_ENV
```

**Comment by amontoison** [comment]
@nimgould 
```
[428/2404] Compiling Fortran object li...p/meson-generated_single_useslls.f90.o
FAILED: libgalahad_single.so.p/meson-generated_single_useslls.f90.o libgalahad_single.so.p/galahad_useslls_single.mod 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_useslls.f90.o -c libgalahad_single.so.p/single_useslls.f90
../src/slls/useslls.F90:392:65:

  392 |             IF ( ABS( prob%X( i ) < ten * stopr ) state = 'LOWER'
      |                                                                 1
Error: Missing ‘)’ in statement at or before (1)
../src/slls/useslls.F90:400:50:

  400 |         nfixed = COUNT( prob%X( : n ) < ten * stop )
      |                                                  1
Error: Symbol ‘stop’ at (1) has no IMPLICIT type; did you mean ‘stopr’?
```

**Comment by jfowkes** [comment]
@amontoison fixed in master

**Comment by amontoison** [comment]
@jfowkes 
```
FAILED: libgalahad_single.so.p/meson-generated_single_useslls.f90.o libgalahad_single.so.p/galahad_useslls_single.mod 
gfortran -Ilibgalahad_single.so.p -I. -I.. -Iinclude -I../include -I../src/dum/include -Isrc/ampl -I../src/ampl -I/home/alexis/Bureau/git/CUTEst/modules/pc64.lnx.gfo/single -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -O0 -g -fopenmp -fPIC -DGALAHAD_SINGLE -DSPRAL_SINGLE -Jlibgalahad_single.so.p -o libgalahad_single.so.p/meson-generated_single_useslls.f90.o -c libgalahad_single.so.p/single_useslls.f90
../src/slls/useslls.F90:392:22:

  392 |             IF ( ABS( prob%X( i ) < ten * stopr ) ) state = 'LOWER'
      |                      1
Error: ‘a’ argument of ‘abs’ intrinsic at (1) must have a numeric type
[432/2346] Compiling Fortran object libgalahad_single.so.p/meson-generated_single_useqpc.f90.o
ninja: build stopped: subcommand failed.
```

**Comment by nimgould** [comment]
Sorry, I really should concentrate, i'm deep in the guts of another debug at present! :(

**Comment by amontoison** [comment]
I try to setup CI with CUTEst such that you can iterate without me but it seems that you changed the sequence of inputs to compile CUTEst (rhhhh).

**Comment by nimgould** [comment]
I haven't touched cutest for months ...

**Comment by nimgould** [comment]
I am confused, the run* should never be part of the shared libraries

**Comment by nimgould** [comment]
The run* file plus the cutest evaluation subroutines need to be compiled and linked with the remaining galahad/cutest libraries at run time

**Comment by nimgould** [comment]
And actually also the use* as these depend on the specific cutest evaluation suboutines for the current problem. It's a two-stage process, first choose a test example, run the SIF decoder to produce the elfuns, groups, ranges routines, then compile and link these with the rest

**Comment by nimgould** [comment]
The galahad sdgal/gal scripts are used here, but these depend on the compiler used and its choices

**Pull request closed without merging**